tests/unit/test_reservation.py

```diff
@@ -0,0 +1,224 @@
+from typing import List
+from unittest.mock import Mock
+
+import pytest
+
+from ni_measurementlink_service._internal.stubs import session_pb2
+from ni_measurementlink_service._internal.stubs.ni.measurementlink.sessionmanagement.v1 import (
+    session_management_service_pb2,
+)
+from ni_measurementlink_service.session_management import (
+    MultiSessionReservation,
+    SessionInformation,
+)
+from tests.utilities import fake_driver
+
+
+def test___single_session_info___create_session___yields_session_info(
+    session_management_client: Mock,
+) -> None:
+    reservation = MultiSessionReservation(
+        session_management_client, _create_grpc_session_infos(1, "nifake")
+    )
+
+    with reservation.create_session(_construct_session, "nifake") as session_info:
+        assert session_info.session_name == "MySession0"
+        assert session_info.resource_name == "Dev0"
+        assert session_info.instrument_type_id == "nifake"
+
+
+def test___single_session_info___create_session___session_created_and_closed(
+    session_management_client: Mock,
+) -> None:
+    reservation = MultiSessionReservation(
+        session_management_client, _create_grpc_session_infos(1, "nifake")
+    )
+
+    with reservation.create_session(_construct_session, "nifake") as session_info:
+        assert isinstance(session_info.session, fake_driver.Session)
+        assert session_info.session.resource_name == "Dev0"
+        assert not session_info.session.is_closed
+
+    assert session_info.session.is_closed
+
+
+def test___single_session_info___create_session___session_cached(
```

Comment:
Since the final assert is that the _session_cache is empty, consider expanding the when clause like `session_cached_for_lifetime_of_create_session` (too wordy but hopefully you get the idea). Unless I'm misunderstanding the cache or test.

Reply:
Renamed to `_session_lifetime_tracked` and moved the `is_closed` check into this test case.